analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.Implementation.cs

```diff
@@ -48,6 +47,20 @@ public KnownType(string fullName, params string[] genericParameters)
         public bool Matches(ITypeSymbol symbol) =>
             IsMatch(symbol) || IsMatch(symbol.OriginalDefinition);
 
+        public override string ToString()
```

Comment:
Should we add documentation comment that this is meant to be used for debugging purposes only? To avoid the temptation of using it in a rule. Maybe with a note that it doesn't support VB.NET version, to explain the reason.

Reply:
We might opt top make it a read-only property `internal string DebuggerDisplay => ..`, and call that one in the `[DebuggerDisplay("{DebuggerDisplay}")]` attribute?

Reply:
Is that testable?

Reply:
Indeed, I think I'll go with the property to make things clear.

Reply:
> _Edit:_ No need to answer 🤦 I just needed to ask loud. It's Monday 😆

Try some caffeine.